models/repo_generate.go

```diff
@@ -210,3 +330,36 @@ func GenerateIssueLabels(ctx DBContext, templateRepo, generateRepo *Repository)
 	}
 	return nil
 }
+
+func generateExpansion(src string, templateRepo, generateRepo *Repository) string {
+	return os.Expand(src, func(key string) string {
```

Comment:
If I read [this](https://golang.org/pkg/os/#Expand) correctly, `os.Expand()` will change both `$REPO_NAME` and `${REPO_NAME}`. It will also drop `${}`. Is that OK?

So, reading the source for `os.Expand()` I think we're better off with [`strings.Replacer`](https://progolang.com/string-replacements-in-go/).

Reply:
Reviewing os.Expand docs I don't see a way to escape possible replacements such that ${REPO_NAME} ends up as $REPO_NAME} in the destination repo.

Reply:
@techknowlogick Letting your imagination go wild... What would you see as proper escapement?

1. `$${var}`
2. `\${var}`
3. `$_{var}`
4. ?

I think `#3` is easy to implement with `RegExp`. The others are difficult because we'd need to go backwards to tell `\${var}` from `\\${var}` from `\\\${var}`, etc.

Reply:
https://play.golang.org/p/bibrkPyM_o_k

Reply:
That's good. My problem is that unknown tokens will produce [_some_ conversion](https://play.golang.org/p/zjUwl8St0Aa) (i.e. removal of either `$` or `${}`). I'd rather have them:

* Neutral (`$UNKNOWN` and `${UNKNOWN}` should remain `$UNKNOWN` and `${UNKNOWN}`)
* Resolve to empty (`"$UNKNOWN"` should resolve to `""`)
* Show some error (e.g. replace `key` with `template error`, but that's problematic, so the error should be returned to the user).

The reason I think like this is that `$` is a very common character for so many languages. A README won't be a problem, but hooks (which are shell and may have awk, sed, regexp embedded) will most certainly have some `$` here and there. It's very error prone and doesn't seem practical.

At this point I wonder if golang templates aren't more convenient 🤔. They're well documented and are quite expandable. They also permit control structures like `if` etc.

Reply:
Perhaps variable expansion could be an optional flag then?  
Go templates would be very powerful, however users would need to learn how to work with them and they leave much more room for errors.  

Between the `.giteatemplate` file and/or a flag, I think accidental expansions would come down to fixing it on the template repo in some form or another and moving on.